### ISSUE TITLE
Upon GDM restart, GUI comes up with login screen but unable to enter into the session

### ISSUE THREAD

**Issue by sachinmokashi** [clearlinux/distribution#673]
**Description**
When I restart the gdm service, I would expect the GUI to come-up normally to the user session

**To Reproduce**

1.  Boot with the Live Desktop Image
2. Open the terminal, type     sudo systemctl restart gdm
3.  systemctl status gdm.path
● gdm.path - Rise GDM service automatically if a graphic card exist
   Loaded: loaded (/usr/lib/systemd/system/gdm.path; static; vendor preset: disabled)
   Active: active (running) since Mon 2019-05-06 03:31:36 UTC; 4h 5min ago

May 06 03:31:36 clr-14a175b6f86f4e6ea94bc31421cfd858 systemd[1]: Started Rise GDM service automatically if a graphic card exist.

4.

**Current behavior**
Upon gdm restart, display goes blank for few seconds, comes-up with the login screen. Upon login, not able to enter the user session, as the UI shows blank

**Expected behavior**
GUI should go blank for few seconds, then should be able to come-up normally

**Environment**:

- Clear Linux OS version: [cat /usr/lib/os-release]
NAME="Clear Linux OS"
VERSION=1
ID=clear-linux-os
ID_LIKE=clear-linux-os
VERSION_ID=29090
PRETTY_NAME="Clear Linux OS"
ANSI_COLOR="1;35"
HOME_URL="https://clearlinux.org"
SUPPORT_URL="https://clearlinux.org"
BUG_REPORT_URL="mailto:dev@lists.clearlinux.org"
PRIVACY_POLICY_URL="http://www.intel.com/privacy"

- journalctl -f attached
[gdm-service-restart-fail.log](https://github.com/clearlinux/distribution/files/3146908/gdm-service-restart-fail.log)

**Comment by mrkz** [comment]
probably related to automatic user login, also, not sure if upstream issue, as I just reproduced this too using a Fedora 27 live image.

**Comment by bwarden** [comment]
I've reproduced this, and was able to switch back to the session on vt2 without actually logging back in.

**Comment by sachinmokashi** [comment]
@mrkz Yes, the session still remains open on TTY2. I tried creating a passwd just to check if automatic signing-in is failing. But even after keying-in the passwd, could not enter session and UI remains blank. However TTY2 still works!

**Comment by bwarden** [comment]
From [this lengthy discussion](https://gitlab.gnome.org/GNOME/gdm/issues/222), it seems like this is behaving pretty much how GNOME intends -- restarting the greeter brings it up in a new X session on a separate VT. I'm inclined to leave this as-is because it's an unusual and intentional scenario, and I'm not convinced the current behavior is strictly wrong. It might be worth considering as an enhancement, though, that logging in again on the new greeter returns you to your already-existing session. I just don't know whether that's possible without major changes upstream.

**Comment by mrkz** [comment]
@bwarden we could open an enhancement request on GDM and let upstream decide on what the default behavior should be, doesn't sound bad indeed, but unsure of the effort required from the DM side.
